pkg/controller/deployment/deployment_controller.go

```diff
@@ -450,6 +451,11 @@ func (dc *DeploymentController) syncRollingUpdateDeployment(deployment extension
 		return dc.updateDeploymentStatus(allRCs, newRC, deployment)
 	}
 
+	// Cleanup old rcs
+	if deployment.Spec.RevisionHistoryLimit != nil {
```

Comment:
> Do you mean add a gc routine to delete the old RCs ?

No need for goroutines, at least in the first place. Right now, you have baked the cleanup policy in the rolling strategy so you will have to reimplement it for recreate too. What I am saying is separate concerns and have the cleanup policy run after the strategy. But you would need to relist the old controllers in order to delete them.